### PR TITLE
Validate configured plugin options the Gatsby way

### PR DESCRIPTION
**Pull request by olavea**

### CODE REVIEW

**Review by raae — COMMENTED** [review]
praise: great work 🥳

File: gatsby-node.js [hunk 1]
```diff
@@ -3,6 +3,23 @@ const { newCloudinary, getResourceOptions } = require('./utils');
...
 
+// Validation of configured plugin options the Gatsby way
+
+exports.pluginOptionsSchema = ({ Joi }) => {
+  return Joi.object({
+    cloudName: Joi.string().required(),
+    apiKey: Joi.string().required(),
+    apiSecret: Joi.string().required(),
+    resourceType: Joi.string().default('image'),
+    resourceType: Joi.string().default('image'),
```
Comment (raae) [inline gatsby-node.js]: nitpick: remove double entry

**Review by raae — CHANGES_REQUESTED** [review]
suggestion(blocking): add `resultsPerPage` and make the default value be the value configured for `maxResult`

This will make it possible to get rid of the logic in `utils -> getResourceOptions` and only to the transformation from camelCase to snake_case there.

I think this can be done by using a [reference](https://joi.dev/api/?v=17.6.0#refkey-options). So something like `.default(Joi.ref("maxResult"))` - mind you I have not tested this.

File: gatsby-node.js [hunk 2]
```diff
@@ -3,6 +3,23 @@ const { newCloudinary, getResourceOptions } = require('./utils');
...
 
+// Validation of configured plugin options the Gatsby way
+
+exports.pluginOptionsSchema = ({ Joi }) => {
+  return Joi.object({
+    cloudName: Joi.string().required(),
+    apiKey: Joi.string().required(),
+    apiSecret: Joi.string().required(),
+    resourceType: Joi.number().integer().positive().default(10),
+    resultsPerPage: Joi.string().default('image'),
```
Comment (raae) [inline gatsby-node.js]: suggestion(blocking): It should be a number like max results and default should be set to the value of max result. See my comment above on how to achieve that.

**Review by raae — CHANGES_REQUESTED** [review]

File: gatsby-node.js (hunk 2 above)
Comment (raae) [reply gatsby-node.js]: Oh I see now that you switched up `resourceType` and `resultsPerPage`.

**Review by raae — CHANGES_REQUESTED** [review]

**Comment by olavea** [comment]
How do we test it?

**Review by raae — COMMENTED** [review]
praise: looks good, I'll give it go and add jest tests before approving

**Comment by raae** [comment]
> How do we test it?

The best is automated Jest tests, I'll add it to this repo after rebasing to main for you.

**Comment by raae** [comment]
Closing this, you can open a PR towards cloudinary-dev

**Pull request closed without merging**